doc/source/drivers/raster/gtiff.rst

```diff
@@ -1094,6 +1094,12 @@ the default behavior of the GTiff driver.
       always written. If set to NO, then the transformation will not be written in
       any situation.
 
+-  .. config:: CHECK_DISK_FREE_SPACE
+      :choices: YES, NO
+      :default: YES
+
+      If ``YES``, ensure that enough disk space is available before attempting to
+      write an uncompressed GeoTIFF.
```

Comment:
Yes, it's not clear to me why someone would need this. I thought it was worth adding because it seems to be used in the wild (https://gis.stackexchange.com/search?q=CHECK_DISK_FREE_SPACE) and it's not obvious from the name that it only applies to GeoTIFF, for example. Maybe qualify its usage a bit further?

If ``YES``, ensure that enough disk space is available before attempting to
write an uncompressed, non-sparse GeoTIFF. Disabling this check is not
expected to be necessary, unless GDAL is incorrectly determining the disk space
available on the destination filesystem.

Reply:
revised in c6c43a805761fe15617ef0c1c77bee58bb48e207